packages/phoenix-evals/src/phoenix/evals/models/mistralai.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from phoenix.evals.models.base import BaseModel
+from phoenix.evals.models.rate_limiters import RateLimiter
+
+if TYPE_CHECKING:
+    from mistralai.models.chat_completion import ChatMessage
+
+DEFAULT_MISTRAL_MODEL = "mistral-large-latest"
+"""Use the latest large mistral model by default."""
+
+MINIMUM_MISTRAL_VERSION = "0.0.11"
+
+
+@dataclass
+class MistralAIModel(BaseModel):
+    """
+    A model class for Mistral AI.
+    Requires mistralai package to be installed.
+    """
+
+    model: str = DEFAULT_MISTRAL_MODEL
+    temperature: float = 0
+    top_p: Optional[float] = None
+    random_seed: Optional[int] = None
+    safe_mode: bool = False
+    safe_prompt: bool = False
+
+    def __post_init__(self) -> None:
+        self._init_client()
+        self._init_rate_limiter()
+
+    @property
+    def _model_name(self) -> str:
+        return self.model
+
+    def _init_client(self) -> None:
+        try:
+            from mistralai.async_client import MistralAsyncClient
+            from mistralai.client import MistralClient
+            from mistralai.exceptions import MistralAPIException
+            from mistralai.models.chat_completion import ChatMessage
+        except ImportError:
+            self._raise_import_error(
+                package_name="mistralai",
+                package_min_version=MINIMUM_MISTRAL_VERSION,
+            )
+        self._client = MistralClient()
+        self._async_client = MistralAsyncClient()
+        self._ChatMessage = ChatMessage
+        self._MistralAPIException = MistralAPIException
+
+    def _init_rate_limiter(self) -> None:
+        self._rate_limiter = RateLimiter(
+            rate_limit_error=self._MistralAPIException,
```

Comment:
This will throttle on too many different exception types and will not properly retry as expected, there's a workaround we can use I'll outline below

Reply:
Cool, let's do that